parser.go

```diff
@@ -478,3 +518,154 @@ func (p *Parser) skipStrings(n int) error {
 
 	return nil
 }
+
+// https://github.com/RedisBloom/RedisBloom/blob/21a2620e75873353fead8c5d70950d3791e36b18/src/rebloom.c#L1116-L1131
+func readBloomFilter(r byteReader) error {
+	// size
+	_, err := redisModuleReadUnsigned(r)
+	if err != nil {
+		return err
+	}
+
+	// numFilters
+	numFilters, err := redisModuleReadUnsigned(r)
+	if err != nil {
+		return err
+	}
+
+	// options
+	_, err = redisModuleReadUnsigned(r)
+	if err != nil {
+		return err
+	}
+
+	// growth
+	_, err = redisModuleReadUnsigned(r)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < numFilters; i++ {
+		// entries
+		_, err = redisModuleReadUnsigned(r)
+		if err != nil {
+			return err
+		}
+
+		// error
+		_, err = redisModuleReadDouble(r)
+		if err != nil {
+			return err
+		}
+
+		// hashes
+		_, err = redisModuleReadUnsigned(r)
+		if err != nil {
+			return err
+		}
+		// bpe
+		_, err = redisModuleReadDouble(r)
+		if err != nil {
+			return err
+		}
+
+		// bits
+		_, err = redisModuleReadUnsigned(r)
+		if err != nil {
+			return err
+		}
+		// n2
+		_, err = redisModuleReadUnsigned(r)
+		if err != nil {
+			return err
+		}
+
+		// string buffer
+		_, err = redisModuleReadStringBuffer(r)
+		if err != nil {
+			return err
+		}
+
+		// size
+		_, err = redisModuleReadUnsigned(r)
+		if err != nil {
+			return err
+		}
+	}
+
+	eof, _, err := readLengthWithEncoding(r)
+	if err != nil {
+		return err
+	}
+
+	if eof != rdbModuleOpcodeEOF {
+		return errors.New(fmt.Sprintf("illegal rdbModuleOpcodeEOF %d,expect:%d", eof, rdbModuleOpcodeEOF))
```

Comment:
You should use `fmt.Errorf` or create a new error type.

Reply:
fixed